### PR TITLE
Add username and timestamp to comment

### PR DESCRIPTION
**Pull request by jyx11011**
Let user input username
Add timestamp to comment
Let user input username to filter comments

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -75,27 +75,59 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
...
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     
     //Set limit options
     FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
     Optional<Integer> limit = getLimit(request);
     if (limit.isPresent()) {
       fetchOptions = FetchOptions.Builder.withLimit(limit.get());
     }
+    
+    //Set username filter
```
Comment (SuyashLakhotia) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Similar comment to the other PR. Format these comments with one space between `//` and the comment text.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -75,27 +75,59 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
...
-      String id = (String) entity.getProperty("id");
-      Comment comment = new Comment(id, content);
+      Comment comment = getCommentFromEntity(entity);
       comments.add(comment);
     }
     return comments;
   }
 
+  /** Returns a Comment object constructed from the given entity. */
+  private Comment getCommentFromEntity(Entity entity) {
```
Comment (SuyashLakhotia) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: "get" usually implies retrieving the data from some other source. A better name might be `transformCommentToEntity`.

Same for the method below that does the opposite.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -76,8 +76,12 @@ function fetchComments(limit = '') {
     });
 }
 
+function getRequestParameter(limit, username) {
+  return 'limit=' + limit + (username != null ? '&username=' + username : '');
```
Comment (SuyashLakhotia) [inline portfolio/src/main/webapp/script.js]: Please see comment I made about `undefined` and `null` in #18.

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (jyx11011) [reply portfolio/src/main/webapp/script.js]: It works for checking undefined. Is it better to use typeof?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -99,8 +99,8 @@ function createElementForComment(comment) {
   contentElement.innerText = comment.content;
   const deleteButton = document.createElement('button');
   deleteButton.innerText = 'Delete';
-  deleteButton.addEventListener('click', () => deleteCommentWithId(comment.id));
-  commentElement.append(userNameElement, timestampElement, contentElement, deleteButton);
+  deleteButton.addEventListener('click', () => deleteComment(comment.id));
+  commentElement.append(userNameElement, contentElement, deleteButton);
```
Comment (SuyashLakhotia) [inline portfolio/src/main/webapp/script.js:103]: Is the removal of timestampElement here intentional?

**Review by SuyashLakhotia — APPROVED** [review]

**Pull request merged**